### PR TITLE
Tag DiffEqCallbacks.jl v0.2.0

### PR DESCRIPTION
**Pull request by attobot**
Repository: [JuliaDiffEq/DiffEqCallbacks.jl](https://github.com/JuliaDiffEq/DiffEqCallbacks.jl)
Release: [v0.2.0](https://github.com/JuliaDiffEq/DiffEqCallbacks.jl/releases/tag/v0.2.0)
Travis: [Travis Build Status](https://travis-ci.org/JuliaDiffEq/DiffEqCallbacks.jl/builds/235495199)
Diff: [vs v0.1.1](https://github.com/JuliaDiffEq/DiffEqCallbacks.jl/compare/3d6d9377f817d3c2bb53bafa78bfe7407697f4c3...d1eb3338949bd2c87ce7f3e15c341cc0d74ef4fa)
`requires` vs v0.1.1: 
```diff
--- 0.1.1/requires
+++ 0.2.0/requires
@@ -1,5 +1,5 @@
 julia 0.5
 DiffEqBase 0.6.0
 NLsolve
-ForwardDiff 0.3.0
+ForwardDiff 0.5.0
 DiffBase
```
cc: @ChrisRackauckas

Please make sure that:
- CI passes for supported Julia versions (if applicable).
- Version bounds are up to date.

### CODE REVIEW

File: DiffEqCallbacks/versions/0.2.0/requires [hunk 1]
```diff
@@ -0,0 +1,5 @@
+julia 0.5
+DiffEqBase 0.6.0
+NLsolve
+ForwardDiff 0.5.0
```
Comment (tkelman) [inline DiffEqCallbacks/versions/0.2.0/requires]: this requirement can only be met on 0.6

**Comment by tkelman** [comment]
Looks like this would cause a failure in DifferentialEquations?
```
ERROR: LoadError: LoadError: LoadError: syntax: invalid macro definition
Stacktrace:
 [1] include_from_node1(::String) at ./loading.jl:552
 [2] include(::String) at ./sysimg.jl:14
 [3] include_from_node1(::String) at ./loading.jl:552
while loading /mnt/pkgevalempty/e2c7c586001b3bb3eeb863b5d7658a596e9096dd/DifferentialEquations/.julia/v0.6/DifferentialEquations/src/fem/fem_heatintegrators.jl, in expression starting on line 41
```

**Comment by ChrisRackauckas** [comment]
>DifferentialEquations/src/fem/fem_heatintegrators.jl

That file hasn't existed in almost 9 months.

**Comment by tkelman** [comment]
Ah, I think I see what's happening. This is interacting in a strange way with https://github.com/JuliaLang/METADATA.jl/pull/9437. JuliaCIBot is cloning this release of DiffEqCallbacks which it maybe shouldn't do directly, it should probably let Pkg resolution respect any bounds that may exist. There would be a conflict where this latest DiffEqCallbacks requires ForwardDiff 0.5.0 but FiniteElementDiffEq and StochasticDiffEq are now upper-bounded. Instead, Pkg is being overly clever and downgrading DifferentialEquations to 0.5.0, which is the most recent version that didn't depend on either FiniteElementDiffEq or StochasticDiffEq. If JuliaCIBot weren't cloning DiffEqCallbacks then this wouldn't happen, instead Pkg would hold DiffEqCallbacks back to 0.1.1, ForwardDiff back to 0.4.2, and use latest DifferentialEquations.

**Comment by ChrisRackauckas** [comment]
Looks like that made the package manager mad:

```julia
julia> Pkg.update()
INFO: Updating METADATA...
INFO: Updating RandomMatrices master...
INFO: Updating Interact master...
INFO: Updating ForwardDiff master...
INFO: Updating ODE master...
INFO: Updating ParameterizedFunctions master...
INFO: Updating GalerkinSparseGrids diffeq...
INFO: Updating DiffEqParamEstim master...
WARNING: Cannot perform fast-forward merge.
INFO: Updating WinRPM master...
INFO: Updating MultiScaleArrays master...
INFO: Updating StochasticDiffEq master...
INFO: Updating PDEOperators master...
INFO: Updating OrdinaryDiffEq master...
INFO: Updating RecursiveArrayTools master...
INFO: Updating DiffEqBase master...
INFO: Computing changes...
ERROR: DifferentialEquations can't be installed because it has no versions that support 0.6.0-rc2.0 of julia. You may need to update METADATA by running `Pkg.update()`
Stacktrace:
 [1] resolve(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{String,Tuple{VersionNumber,Bool}}, ::Dict{String,Base.Pkg.Types.Fixed}, ::Dict{String,VersionNumber}, ::Set{String}) at .\pkg\entry.jl:486
 [2] update(::String, ::Set{String}) at .\pkg\entry.jl:461
 [3] (::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#update,Tuple{String,Set{String}}})() at .\pkg\dir.jl:36
 [4] cd(::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#update,Tuple{String,Set{String}}}, ::String) at .\file.jl:59
 [5] #cd#1(::Array{Any,1}, ::Function, ::Function, ::String, ::Vararg{Any,N} where N) at .\pkg\dir.jl:36
 [6] update() at .\pkg\pkg.jl:228
```

Getting the other two PRs in would be a direct fix for this instance, but maybe it should be taken note of.

**Comment by tkelman** [comment]
do you have things locally checked out?

**Pull request merged**